### PR TITLE
Remove duplicates and fix namespaces in S.T.RegularExpressions.Tests

### PR DESCRIPTION
**Pull request by ViktorHofer**
Relates to https://github.com/dotnet/runtime/pull/33915 (VSTest runner logs duplicate test rows)

### CODE REVIEW

File: src/libraries/System.Text.RegularExpressions/tests/MonoRegexTests.cs [hunk 1]
```diff
@@ -452,8 +449,6 @@ public static IEnumerable<object[]> RegexTestCases()
             yield return new object[] { @"((?-i:a))b", RegexOptions.IgnoreCase, "aB", "Pass. Group[0]=(0,2) Group[1]=(0,1)" };
             yield return new object[] { @"(?-i:a)b", RegexOptions.IgnoreCase, "Ab", "Fail." };
             yield return new object[] { @"((?-i:a))b", RegexOptions.IgnoreCase, "Ab", "Fail." };
-            yield return new object[] { @"(?-i:a)b", RegexOptions.IgnoreCase, "aB", "Pass. Group[0]=(0,2)" };
```
Comment (stephentoub) [inline src/libraries/System.Text.RegularExpressions/tests/MonoRegexTests.cs]: Where is this one duplicated?

Comment (stephentoub) [reply src/libraries/System.Text.RegularExpressions/tests/MonoRegexTests.cs]: Nevermind, found it.

File: src/libraries/System.Text.RegularExpressions/tests/MonoRegexTests.cs [hunk 2]
```diff
@@ -1030,7 +1015,6 @@ public static IEnumerable<object[]> RegexTestCases()
             yield return new object[] { @"^(((?<foo>\()[^()]*)+((?<bar-foo>\))[^()]*)+)+(?(foo)(?!))$", RegexOptions.None, "((a(b))c))","Fail." };
             yield return new object[] { @"^(((?<foo>\()[^()]*)+((?<bar-foo>\))[^()]*)+)+(?(foo)(?!))$", RegexOptions.None, ")(","Fail." };
             yield return new object[] { @"^(((?<foo>\()[^()]*)+((?<bar-foo>\))[^()]*)+)+(?(foo)(?!))$", RegexOptions.None, "((a((b))c)","Fail." };
-            yield return new object[] { @"b", RegexOptions.RightToLeft, "babaaa", "Pass. Group[0]=(2,1)" };
```
Comment (stephentoub) [inline src/libraries/System.Text.RegularExpressions/tests/MonoRegexTests.cs]: Where is this one duplicated?

File: src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs [hunk 3]
```diff
@@ -28,9 +28,6 @@ public static IEnumerable<object[]> Match_Basic_TestData()
             // Testing control character escapes???: "2", "(\u0032)"
             yield return new object[] { "(\u0034)", "4", RegexOptions.None, 0, 1, true, "4", };
 
-            // Using *, +, ?, {}: Actual - "a+\\.?b*\\.?c{2}"
-            yield return new object[] { @"a+\.?b*\.+c{2}", "ab.cc", RegexOptions.None, 0, 5, true, "ab.cc" };
```
Comment (stephentoub) [inline src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs]: Where is this one duplicated?

**Comment by ViktorHofer** [comment]
Forgot to mention that I used a VSCode extension for identifying and trimming most of the duplicates.

File: src/libraries/System.Text.RegularExpressions/tests/MonoRegexTests.cs (hunk 2 above)
Comment (ViktorHofer) [reply src/libraries/System.Text.RegularExpressions/tests/MonoRegexTests.cs]: https://github.com/dotnet/runtime/blob/5e6711dbfb1b696e1fb7c53f9d84d33c60dc87be/src/libraries/System.Text.RegularExpressions/tests/MonoRegexTests.cs#L798

File: src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs (hunk 3 above)
Comment (ViktorHofer) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs]: https://github.com/dotnet/runtime/blob/5e6711dbfb1b696e1fb7c53f9d84d33c60dc87be/src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs#L169

**Review by stephentoub — APPROVED** [review]

**Pull request merged**